custom_metrics/Images.js

```diff
@@ -2,19 +2,26 @@ var wptImages = function(win) {
   var images = [];
   if (win) {
     var doc = win.document;
-    var elements = doc.getElementsByTagName('*');
-    var re = /url\((http.*)\)/ig;
+    var elements = doc.getElementsByTagName("*");
     for (var i = 0; i < elements.length; i++) {
       var el = elements[i];
-      if (el.tagName == 'IMG') {
+      if (el.tagName == "IMG") {
         var url = el.currentSrc || el.src;
 
         // Only include HTTP(S) URLs i.e. skip dataURIs
-        if(url.indexOf("http") === 0) {
-          images.push({'url': url, 'width': el.width, 'height': el.height, 'naturalWidth': el.naturalWidth, 'naturalHeight': el.naturalHeight});
+        if (url.indexOf("http") === 0) {
+          images.push({
+            url: url,
+            width: el.width,
+            height: el.height,
+            naturalWidth: el.naturalWidth,
+            naturalHeight: el.naturalHeight,
+            loading: el.getAttribute("loading"),
+            "in-viewport": el.getBoundingClientRect().top < window.innerHeight,

```

Comment:
Would you also like to add a property that combines both values, for example
```suggestion
            "in-viewport": el.getBoundingClientRect().top < window.innerHeight,
            lazyAboveFold: el.getAttribute("loading") == "lazy" && el.getBoundingClientRect().top < window.innerHeight
```

Alternatively, you can evaluate that in the SQL but this would save a step in the analysis.

Reply:
Personally I thought it was better to evaluate this in SQL to avoid having multiple metrics (or a combination of them) giving the same information (wasteful) and we may want to change/extend what we are reporting. However I think you are in a better position to know which is the better option.

Reply:
SGTM